farm.js

```diff
@@ -1,18 +1,31 @@
 #!/usr/bin/env node
 'use strict';
 
+const Bird = require('bird-foo');
 const Cow = require('cow-foo');
 const feedCow = require('cow-foo/feed-cow');
 
+const feedBird = require('./lib/feed-bird');
 const eatLunch = require('./lib/eat-lunch');
 
 
 function main(cb) {
   const cow = new Cow();
   feedCow(cow, function() {
-    eatLunch(function() {
-      cb();
-    });
+    var bird = new Bird();
+    if(bird == 'pigeon') {
+      feedBird(bird, 'seed', function() {
+        eatLunch(function() {
+          cb();
```

Comment:
Lets defer the callback > promise refactor and do that after this pull request has landed.

I've opened issue to document that work: https://github.com/yola/farm-foo/issues/2